src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -915,19 +915,21 @@ private void StopWorkerProcess()
                 return;
             }
 
+            int gracePeriod = WorkerConstants.WorkerTerminateGracePeriodInSeconds;
+
             var workerTerminate = new WorkerTerminate()
             {
-                GracePeriod = Duration.FromTimeSpan(TimeSpan.FromSeconds(WorkerConstants.WorkerTerminateGracePeriodInSeconds))
+                GracePeriod = Duration.FromTimeSpan(TimeSpan.FromSeconds(gracePeriod))
             };
 
-            _workerChannelLogger.LogDebug($"Sending WorkerTerminate message with grace period {WorkerConstants.WorkerTerminateGracePeriodInSeconds} seconds.");
+            _workerChannelLogger.LogDebug($"Sending WorkerTerminate message with grace period {gracePeriod} seconds.");
 
             SendStreamingMessage(new StreamingMessage
             {
                 WorkerTerminate = workerTerminate
             });
 
-            WorkerProcess.WaitForProcessExit();
+            WorkerProcess.WaitForProcessExitInMilliSeconds(gracePeriod * 1000);
```

Comment:
super nit here, but I don't feel we need to change the name, the method definition/summary should be enough to indicate if the parameter it takes is milliseconds or second